From 4e6279ef8230ee515b069a1dde542e49a0ca5b43 Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Mon, 28 Apr 2025 02:04:11 +0200
Subject: [PATCH 068/115] drm/panfrost: Add support for Mali G76 MP4 on the
 Mediatek MT6785

MediaTek MT6785 has a Mali-G76 MC4 (Bifrost 3rd generation),
add a new compatible and platform data with four power domains.

Signed-off-by: Dinolek <git@dinolek.me>
---
 drivers/gpu/drm/panfrost/panfrost_drv.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/panfrost/panfrost_drv.c b/drivers/gpu/drm/panfrost/panfrost_drv.c
index 784e36d72..6f06c94b0 100644
--- a/drivers/gpu/drm/panfrost/panfrost_drv.c
+++ b/drivers/gpu/drm/panfrost/panfrost_drv.c
@@ -1093,6 +1093,15 @@ static const struct panfrost_compatible amlogic_data = {
 
 static const char * const mediatek_pm_domains[] = { "core0", "core1", "core2",
 						    "core3", "core4" };
+
+static const struct panfrost_compatible mediatek_mt6785_data = {
+	.num_supplies = ARRAY_SIZE(default_supplies) - 1,
+	.supply_names = default_supplies,
+	.num_pm_domains = 4,
+	.pm_domain_names = mediatek_pm_domains,
+	.pm_features = BIT(GPU_PM_CLK_DIS) | BIT(GPU_PM_VREG_OFF),
+};
+
 /*
  * The old data with two power supplies for MT8183 is here only to
  * keep retro-compatibility with older devicetrees, as DVFS will
@@ -1171,6 +1180,7 @@ static const struct of_device_id dt_match[] = {
 	{ .compatible = "arm,mali-t880", .data = &default_data, },
 	{ .compatible = "arm,mali-bifrost", .data = &default_data, },
 	{ .compatible = "arm,mali-valhall-jm", .data = &default_data, },
+	{ .compatible = "mediatek,mt6785-mali", .data = &mediatek_mt6785_data },
 	{ .compatible = "mediatek,mt8183-mali", .data = &mediatek_mt8183_data },
 	{ .compatible = "mediatek,mt8183b-mali", .data = &mediatek_mt8183_b_data },
 	{ .compatible = "mediatek,mt8186-mali", .data = &mediatek_mt8186_data },
-- 
2.56.0

